packages/tests/server/react/useQuery.test.tsx

```diff
@@ -311,6 +311,33 @@ describe('useQuery()', () => {
       ]
     `);
   });
+
+  test('using queryOptions', async () => {
+    const { client, App } = ctx;
+    function MyComponent() {
+      const queryOptions = client.post.byId.useQueryOptions({ id: '1' });
+      expect(queryOptions.trpc.path).toBe('post.byId');
+      const query1 = useQuery(queryOptions);
```

Comment:
Niiiiiiceeeeeeeee

Reply:
Will this work with the `trpc: {}` props as well?

Reply:
Probably with some modifications.

This was very much a first draft. I think we neee to decide what to do with the tRPC client, if it should be injected from a factory or if react context is still required.

I think @Nick-Lucas doesnt like it requiring react context

Reply:
I wonder if we should do a new `@trpc/tanstack-query` pkg (and maybe `@trpc/tanstack`) and deprecate `@trpc/react-query` if we go this direction 🤔 🤔 we could probably make a codemod that does the migration 

Would be so nice with a clean slate